### PR TITLE
Apim 2019-01-01

### PR DESCRIPTION
**Pull request by solankisamir**
- Update resource manager schema for 2019-01-01 api version 
- Add tests

### CODE REVIEW

**Comment by solankisamir** [comment]
@Tiano2017 can you take a look?

**Comment by solankisamir** [comment]
@Tiano2017 a gentle ping on this!

**Comment by Tiano2017** [comment]
@bmoore-msft do you want to take a look at this PR? it's adding 30+ new resource types.

**Comment by solankisamir** [comment]
@bmoore-msft  ping!

**Comment by bmoore-msft** [comment]
Tian and I had a quick conversation, the # of additions is fine for the top-level schema.

**Comment by solankisamir** [comment]
@Tiano2017 a ping on this

File: schemas/2015-01-01/deploymentTemplate.json [hunk 1]
```diff
@@ -470,6 +470,42 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2018-06-01-preview/Microsoft.ApiManagement.json#/resourceDefinitions/service_tags" },
                   { "$ref": "https://schema.management.azure.com/schemas/2018-06-01-preview/Microsoft.ApiManagement.json#/resourceDefinitions/service_templates" },
                   { "$ref": "https://schema.management.azure.com/schemas/2018-06-01-preview/Microsoft.ApiManagement.json#/resourceDefinitions/service_users" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2019-01-01/Microsoft.ApiManagement.json#/resourceDefinitions/service" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >{ "$ref": "https://schema.management.azure.com/schemas/2019-01-01/Microsoft.ApiManagement.json#/resourceDefinitions/service" }, [](start = 18, length = 127)

we have a new root schema now: 2019-04-01/deploymentTemplate.json. Can you please add them to the new schema as well?

**Comment by solankisamir** [comment]
@TianoMS ping on this !

**Comment by solankisamir** [comment]
@TianoMS the documentation got update already.... and the schema is not published.
https://docs.microsoft.com/en-us/azure/templates/microsoft.apimanagement/2019-01-01/service

**Pull request merged**